app/controllers/edit_phone_controller.rb

```diff
@@ -2,12 +2,21 @@ class EditPhoneController < ApplicationController
   include ChangeCoreCredentials
 
   before_action :authenticate_user!
-  before_action :enforce_has_phone!
   before_action :enforce_recent_mfa!
 
   helper_method :resource
+  helper_method :setting_up_first_phone?
+  helper_method :start_path
+
+  def create
+    redirect_to edit_user_registration_phone_path unless setting_up_first_phone?
```

Comment:
🤔 I'm wondering if this could be a before action.

Pro: Helps make this controller even leaner
Con: This is quite clear, whereas you'd need to go away and follow up a before method that is a little hidden at the top, plus we'd need two of them one for L12 other for L18...

@AlanGabbianelli what would your expectation be here based on the lean controller conversation we were having in slack?

Reply:
I generally like before actions but in this case, I don't think it would make the controller leaner or clearer. The two redirects in `new` and and `show` are very clear as they are. Reading them, we can understand that if we end up in the wrong action (based on the boolean value of `setting_up_first_phone?`), we'll be redirected to the other.